[Synthetics] Add kibana space id to documents

shahzad31 opened the pull request:
## Summary

Add kibana space id to documents !!

Field is added as `meta.space_id`

<img width="1728" alt="image" src="https://github.com/elastic/kibana/assets/3505601/c11f9dfb-93c9-4f32-a862-99a5043c8832">

shahzad31 committed:
add space id to documents

elasticmachine commented:
Pinging @elastic/uptime (Team:uptime)

apmmachine commented:
### :robot: GitHub comments

<details><summary>Expand to view the GitHub comments</summary>
<p>

Just comment with:
- `/oblt-deploy` : Deploy a Kibana instance using the Observability test environments.
- `/oblt-deploy-serverless` : Deploy a serverless Kibana instance using the Observability test environments.
- `run` `elasticsearch-ci/docs` : Re-trigger the docs validation. (use unformatted text in the comment!)

</p>
</details>

vitaliidm committed:
[Security Solution][Detection Engine] improves new terms rule for multiple fields (#157413)

## Summary

As described in our README for new terms rule type:

> Runtime field supports only 100 emitted values. So for large arrays or
combination of values greater than 100, results may not be exhaustive.
This applies only to new terms with multiple fields.
  Following edge cases possible:
- false negatives (alert is not generated) if too many fields were
emitted and actual new values are not getting evaluated if it happened
in document in rule run window.
- false positives (wrong alert generated) if too many fields were
emitted in historical document and some old terms are not getting
evaluated against values in new documents.

To avoid this and deliver the better experience for our customers, this
PR is moving from current implementation(emitting aggregated values for
multiple new terms fields) towards using composite aggregation for each
page from phase 1, split in chunks by 500.
This allowed to be done due
[order](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-bucket-composite-aggregation.html#_order)
of composite aggregation results

NOTE: implementation for a single new terms filed is the same, due to
performance reasons


### Performance measurements

Implementation | Shards | Docs per shard | Simultaneous Rule Executions
| Fields cardinality | Rule Execution Time Runtime field(current
implementation) | On week work
-- | -- | -- | -- | -- | -- | -- 
array of unique values length 10 |   |   |   |   |   |   
Terms 1 field | 10 | 900,000 | 1 | 100,000 | |   
Terms 2 fields | 10 | 900,000 | 1 | 100,000 | 30s  |  41s
Terms 3 fields | 10 | 900,000 | 1 | 100,000 | 40s | 56s

Implementation | Shards | Docs per shard | Simultaneous Rule Executions
| Fields cardinality | Rule Execution Time Runtime field(current
implementation) | On week work 1,000 per batch | On week work 500 per
batch
-- | -- | -- | -- | -- | -- | -- | --
Terms 2 fields | 10 | 9,000,000 | 1 | 100,000 | 19s | 41s | 35s 
Terms 3 fields | 10 | 9,000,000 | 1 | 100,000 | 21s | 52s| 47s 
CPU % | | | | | 400-450% |500-600% | 400-450%

I selected size of the chunk as 500, since it's a bit faster and less
load on CPU

### Considerations on parallel composite search requests in phase 2

When running composite search requests in parallel, noticed significant
CPU increase in Elasticsearch ~ 1,000% for 2 requests in parallel
against ~ 500% for single.
Where win in performance was not that big: ~ 35s for 2 in parallel, 43s
for a single request. I think, having only one request is the better
option to go, that will prevent unnecessary CPU usage

### Test cases
I've added several functional test cases, that ensures, no missing/false
positives alerts are occurring. Applied to the old implementation, they
would fail

### Retry on max_clause_count error
Because we create query, that can have few thousands clauses, it is
possible it may fail due to [the maximum number of allowed
clauses](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-settings.html)
I implemented retry that: If request fails with batch size of 500
(default value), we will try to reduce it in twice per each retried
request, up until 125. Per ES documentation, max_clause_count min value
is 1,000 - so with 125 we should be able execute query below
max_clause_count value

### Checklist

Delete any items that are not applicable to this PR.

- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

leemthompo committed:
[Search] ES3 UI copy updates (#170755)

Fixes some incorrect UI copy

jbudz committed:
[build] Skip CDN assets build when testing docker contexts (#170735)

When testing docker contexts we skip all build steps related to Kibana
artifacts. CDN assets rely on the Kibana base folder and should be
skipped too.

Fixes
https://buildkite.com/elastic/kibana-artifacts-snapshot/builds/3445
Build
https://buildkite.com/elastic/kibana-artifacts-snapshot/builds/3451
Bug introduced in https://github.com/elastic/kibana/pull/169707

shahzad31 committed:
fix types

marshallmain committed:
[Security Solution][Detection Engine] Re-word error message to be more accurate (#170582)

## Summary

The error message we write when rule execution fails references
"indexing of alerts" even if the error is not related to indexing
alerts. Often this happens when errors occur while searching source
documents.
### Before, when some source indices have `@timestamp` mapped
incorrectly:

![image](https://github.com/elastic/kibana/assets/55718608/004ffe94-c9a3-485b-9de7-4150cef2c192)
### After, when some source indices have `@timestamp` mapped
incorrectly:

![image](https://github.com/elastic/kibana/assets/55718608/d68abb27-2e28-4242-8261-41fbbe80327d)

cnasikas committed:
[Cases] Support bulk creating cases though the cases client (#170326)

kibanamachine committed:
skip failing test suite (#170753)

lukeelmers committed:
Add serverless section to dev docs menu. (#170581)

MadameSheema committed:
[Security Solution] Fixes and unskips detection page filters tests (#170763)

nchaulet committed:
[Fleet] Display package root requirements (#170478)

mistic committed:
skip flaky suite (#169159)

mistic committed:
skip flaky suite (#168340)

mistic committed:
skip flaky suite (#170674)

mistic committed:
skip flaky suite (#170794)

mistic committed:
skip flaky suite (#169689)

mistic committed:
skip flaky suite (#169821)

mistic committed:
skip flaky suite (#170604)

mistic committed:
skip flaky suite (#170811)

mistic committed:
skip flaky suite (#170706)

mistic committed:
skip flaky suite (#170812)

mistic committed:
skip flaky suite (#170667)

mistic committed:
chore(NA): update versions after v8.11.1 bump (#170738)

This PR is a simple update of our versions file after the recent bumps.

mistic committed:
skip flaky suite (#170667)

mistic committed:
skip flaky suite (#170601)

kibanamachine committed:
skip failing test suite (#170690)

kibanamachine committed:
[api-docs] 2023-11-08 Daily api_docs build (#170823)

Generated by
https://buildkite.com/elastic/kibana-api-docs-daily/builds/515

criamico committed:
[Fleet] Modify bulk unenroll to include inactive agents (#170249)

Fixes https://github.com/elastic/kibana/issues/169977
## Summary
[Fleet] Modify bulk unenroll to include inactive agents when passing
`"includeInactive": true`.

- This flag only works when the agents are selected by query.
- Updated the hook in the UI to include the new flag.


### Testing

#### API
- Enroll some agents and make them inactive
- From dev tools, run the bulk_unenroll api with option
`includeInactive`:
```
POST kbn:/api/fleet/agents/bulk_unenroll
{
    "agents": "last_checkin<=2023-11-06",
    "revoke": true,
    "includeInactive": true
}
```
This should remove all the agents selected in the kuery, also the
inactive ones.

#### UI
-  Enroll some agents and make them inactive
- Select all agents on all pages, from the bulk actions click on
"unenroll agents"
- It should remove all the agents, inactive ones too.


### Checklist
- [ ]
[Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html)
was added for features that require explanation or tutorials
- [ ] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios


### For maintainers

- [ ] This was checked for breaking API changes and was [labeled
appropriately](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

---------

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

shahzad31 committed:
added migration

shahzad31 committed:
Merge branch 'space-id' of https://github.com/shahzad31/kibana into space-id

shahzad31 committed:
update tests

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/jest_integration -u src/core/server/integration_tests/ci_checks'

sphilipse commented:
@shahzad31 pretty sure you messed up a merge/rebase here
